src/test/java/steps/StepAssertor.java

```diff
@@ -63,13 +63,32 @@ private static void assertTrue(boolean test, String errorMessage) {
         }
     }
 
-    private static boolean testCategoryCheck(String json, String responseObject, String key, String expected) {
-        if (responseObject.startsWith("Category ")) {
-            String jsonPath = "$.categoryChecks..[?(@.calculationType == '" + responseObject + "')]." + KEY_MAP.get(key);
+    private static boolean testHttpResponse(Response response, String responseObject, String key, String expected) {
```

Comment:
There are a few alternative methods that could be used in this class to improve readability.
```
.equalsIgnoreCase()
Integer.toString()
list.isEmpty()
```
Obviously a bit of opinion in this & in a test class so down to your discretion.